test/simple.html

```diff
@@ -0,0 +1,132 @@
+﻿<!doctype html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
+
+    <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+    <script src="../../web-component-tester/browser.js"></script>
+
+    <!-- Step 1: import the element to test -->
+    <link rel="import" href="../juicy-select.html">
+    <script src="./helpers.js"></script>
+</head>
+
+<body>
+
+    <!-- You can use the document as a place to set up your fixtures. -->
+    <test-fixture id="juicy-select-fixture">
+        <template>
+            <!-- nest to workaround test-fixture bug -->
+            <div>
+                <dom-bind>
+                    <template>
+                        <juicy-select caption-text="No favorite country" value="{{model.TempValue}}" options="{{model.Countries}}" selected-property="Selected"
+                            text-property="Name" value-property="Name"></juicy-html>
+                    </template>
+                </dom-bind>
+            </div>
+        </template>
+    </test-fixture>
+
+    <script>
+        describe('<juicy-select>', function (done) {
+            var juicySelect, domBind;
+            const model = {
+                Countries: [{ Name: "Sweden", Selected: false }, { Name: "China", Selected: false }, { Name: "India", Selected: false }],
+            }
+            beforeEach(function () {
+                domBind = fixture('juicy-select-fixture').querySelector('dom-bind');
+                juicySelect = document.querySelector('juicy-select');
+
+                domBind.set('model', model);
+            });
+            it("Should contain correct number of <option> elements: options.length + 1 (for caption)", function () {
+                expect(juicySelect.$select.childNodes.length).to.equal(juicySelect.options.length + 1);
+            });
+            it("Each <option>.innerHTML should equal model.options[index][text-property]", function () {
+                const childNodes = Object.entries(juicySelect.$select.childNodes);
+
+                for (let [index, option] of childNodes) {
+                    index = ~~index;
```

Comment:
Please don't use bitwise sorcery. 🧙 Do you [mean Math.floor](https://stackoverflow.com/questions/5971645/what-is-the-double-tilde-operator-in-javascript)?

Reply:
It's a fast alternative to parseInt. But you're right.